src/project_manager/ros_colcon_step.cpp

```diff
@@ -212,6 +212,18 @@ QString ROSColconStep::makeCommand() const
     }
 }
 
+void ROSColconStep::stdOutput(const QString &line)
+{
+    AbstractProcessStep::stdOutput(line);
+    if (m_percentProgress.indexIn(line, 0) != -1)
```

Comment:
The immediate cause of progress not updating is that the default Colcon output is rather limited compared to CatkinTools. When `line` is passed in here it contains something like `Starting >>> examples_rclcpp_minimal_action_client\n`. The regex up on line 77 fails to parse it since the expected format doesn't match, which means that nothing is emitted at line 223.